contracts/deployables/freeze-voting/FreezeVotingStandaloneV1.sol

```diff
@@ -0,0 +1,519 @@
+    {
+        FreezeVotingStandaloneStorage
+            storage $ = _getFreezeVotingStandaloneStorage();
+        return $.votingConfigs;
+    }
+
+    /**
+     * @inheritdoc IFreezeVotingStandaloneV1
+     */
+    function votingConfig(
+        uint256 index
+    ) public view virtual override returns (IVotingTypes.VotingConfig memory) {
+        FreezeVotingStandaloneStorage
+            storage $ = _getFreezeVotingStandaloneStorage();
+        if (index >= $.votingConfigs.length) {
+            revert InvalidVotingConfig(index);
+        }
+        return $.votingConfigs[index];
+    }
+
+    /**
+     * @inheritdoc IFreezeVotingStandaloneV1
+     */
+    function unfreezeVotesThreshold()
+        public
+        view
+        virtual
+        override
+        returns (uint256)
+    {
+        FreezeVotingStandaloneStorage
+            storage $ = _getFreezeVotingStandaloneStorage();
+        return $.unfreezeVotesThreshold;
+    }
+
+    /**
+     * @inheritdoc IFreezeVotingStandaloneV1
+     */
+    function unfreezeProposalPeriod()
+        public
+        view
+        virtual
+        override
+        returns (uint32)
+    {
+        FreezeVotingStandaloneStorage
+            storage $ = _getFreezeVotingStandaloneStorage();
+        return $.unfreezeProposalPeriod;
+    }
+
+    /**
+     * @inheritdoc IFreezeVotingStandaloneV1
+     */
+    function getUnfreezeProposalVotes()
+        public
+        view
+        virtual
+        override
+        returns (uint256)
+    {
+        FreezeVotingStandaloneStorage
+            storage $ = _getFreezeVotingStandaloneStorage();
+        return $.unfreezeProposalVoteCount;
+    }
+
+    // --- State-Changing Functions ---
+
+    /**
+     * @inheritdoc IFreezeVotingStandaloneV1
+     */
+    function castFreezeVote(
+        IVotingTypes.VotingConfigVoteData[] calldata votingConfigsToUse_,
+        uint256 lightAccountIndex_
+    ) public virtual override {
+        // Check if already frozen
+        if (isFrozen()) revert AlreadyFrozen();
+
+        // Resolve voter address (handles light account)
+        address voter = potentialLightAccountResolvedOwner(
+            msg.sender,
+            lightAccountIndex_
+        );
+
+        // Check if proposal is expired and create new one if needed
+        FreezeVotingBaseStorage storage $base = _getFreezeVotingBaseStorage();
+        if (
+            block.timestamp >
+            $base.freezeProposalCreated + $base.freezeProposalPeriod
+        ) {
+            _initializeFreezeVote();
+            emit FreezeProposalCreated(uint48(block.timestamp), voter);
+        }
+
+        // Calculate total weight from all configs
+        uint256 totalWeight = _getVotes(voter, votingConfigsToUse_);
+
+        // Record the vote and handle freeze activation if threshold is reached
+        _recordFreezeVote(voter, totalWeight);
+    }
+
+    /**
+     * @inheritdoc IFreezeVotingStandaloneV1
+     */
+    function castUnfreezeVote(
```

Comment:
Do we want this to automatically call `unfreeze` if the votes threshold is reached?

Reply:
yeah that's a good callout. i'll make it so that unfreezing happens automatically after unfreeze threshold is reached, to match the pattern when casting freeze votes.